### PR TITLE
Two tests for issue #677 - one green and one red

### PR DESCRIPTION
**Pull request by RogerKratz**
Had compile problems on dev branch so I just added the test file. Feel free to put the tests where they belong.

AFAIK - both tests should be green, no matter if instance is replaced on OnActivating or in module's AttachToComponentRegistration method.

### CODE REVIEW

File: test/Autofac.Test/ReplaceInstanceAndReleaseTest.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            builder.RegisterType<someComponent>()
+                .OnRelease(c => Assert.True(c.IsReplaced));
+
+            using (var container = builder.Build())
+            {
+                container.Resolve<someComponent>();
+            }
+        }
+
+        private class activatingModule : Module
```
Comment (tillig) [inline test/Autofac.Test/ReplaceInstanceAndReleaseTest.cs]: ReplaceInstanceModule

File: test/Autofac.Test/ReplaceInstanceAndReleaseTest.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            protected override void AttachToComponentRegistration(IComponentRegistry componentRegistry, IComponentRegistration registration)
+            {
+                registration.Activating += (o, args) =>
+                {
+                    args.ReplaceInstance(new someComponent { IsReplaced = true });
+                };
+            }
+        }
+
+        private class someComponent
```
Comment (tillig) [inline test/Autofac.Test/ReplaceInstanceAndReleaseTest.cs]: ReplaceableComponent

File: test/Autofac.Test/ReplaceInstanceAndReleaseTest.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
+﻿using Autofac.Core;
+using Xunit;
+
+namespace Autofac.Test
+{
+    public class ReplaceInstanceAndReleaseTest
+    {
+        [Fact]
+        public void ReplaceInstance_ReleaseTestWithOnActivating()
```
Comment (tillig) [inline test/Autofac.Test/ReplaceInstanceAndReleaseTest.cs]: ReplaceInstance_RegistrationActivatingHandlerProvidesResultToRelease

File: test/Autofac.Test/ReplaceInstanceAndReleaseTest.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                .OnRelease(c => Assert.True(c.IsReplaced));
+
+            using (var container = builder.Build())
+            {
+                container.Resolve<someComponent>();
+            }
+        }
+
+        [Fact]
+        public void ReplaceInstance_ReleaseTestWithModuleActivating()
```
Comment (tillig) [inline test/Autofac.Test/ReplaceInstanceAndReleaseTest.cs]: ReplaceInstance_ModuleActivatingHandlerProvidesResultToRelease

File: test/Autofac.Test/ReplaceInstanceAndReleaseTest.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                .OnActivating(c => c.ReplaceInstance(new someComponent { IsReplaced = true }))
+                .OnRelease(c => Assert.True(c.IsReplaced));
+
+            using (var container = builder.Build())
+            {
+                container.Resolve<someComponent>();
+            }
+        }
+
+        [Fact]
```
Comment (tillig) [inline test/Autofac.Test/ReplaceInstanceAndReleaseTest.cs]: `[Fact(Skip="Issue #677")]`

**Comment by tillig** [comment]
Looks like these should go in `test/Autofac.Test/Core/ContainerTests.cs` since that's where the other `ReplaceInstance` tests are.

If you're not interested/able to do that, let me know and we can put the tests as code in the issue rather than as a PR.

**Comment by RogerKratz** [comment]
Ok. Done.
(Now two different commits, don't know if you want me to squash them)

**Comment by tillig** [comment]
You rock! Thanks so much for the help. PR merged.

**Pull request merged**